infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLog.java

```diff
@@ -86,4 +87,22 @@
      * Clears all data and resets all segment handlers.
      */
     void reset();
+
+    default OverwriteCause verifyOverwriteCauseForAddress(long address, LogData entry) {
```

Comment:
Should this really be part of the interface?

Reply:
consider moving this to the logunit class?

Reply:
I thought of making it part of a utils file since I need it both from the StreamLogFiles and the InMemoryStreamLog, but it relies on the read function, so I thought it was maybe better to be part of the base definition and not duplicate code. Or if preferred I can move it to a utils file.